design/decouple-ceph-version.md

```diff
@@ -0,0 +1,88 @@
+osd, rgw, and mds pods. The significance of this approach is that the Rook binary is not included in the daemon containers. All initialization performed by Rook to initialize and generate the Ceph config must be completed in an [init container](https://github.com/rook/rook/issues/2003). Once the Rook init containers complete their execution, the daemon container can run the Ceph image. The daemon container will no longer have Rook running.
+
+In the following Cluster CRD example, the Ceph version is Mimic 13.2.1.
+
+```yaml
+apiVersion: ceph.rook.io/v1beta1
+kind: Cluster
+metadata:
+  name: rook-ceph
+  namespace: rook-ceph
+spec:
+  cephImage: ceph/ceph:v13.2.1
+...
+```
+
+### Operator Requirements
+
+The operator needs to run the Ceph client tools to manage the cluster. For example, the `ceph` tool is needed for general Ceph configuration and status, `radosgw-admin` is required for managing an object store. Therefore, all the necessary client tools will still be included in the Rook image.
+
+The client tools are tested by the Ceph team to be backward and forward compatible by two versions. Therefore, the operator can support a version of Ceph up to two versions older than the client tools it contains. The operator will also be able to orchestrate upgrading to two versions newer of Ceph, in case there is a need to connect the operator to a newer version of a daemon before the operator is updated.
+
+The operator will be made to understand differences in the Ceph versions that are necessary for orchestration. Some examples might include:
+- If running Luminous, start the Ceph dashboard on http. If running Mimic, a self-signed cert could be generated to start the dashboard with https.
+- If a new daemon is added in a future Ceph release, the operator would understand to deploy that daemon only if the Ceph version is at least that version.
+
+### Upgrades
+
+The flexibility during upgrades will now be improved since the upgrade of Rook will be independent from the upgrade to the Ceph version.
+- To upgrade Rook, update the version of the Rook operator container
+- To upgrade Ceph, update the `cephImage` in the cluster CRD
+
+Each time the operator starts, an idempotent orchestration is executed to ensure the cluster is in the desired state. As part of the orchestration, the version of the operator
+will be reviewed. If the version has changed, the operator will update each of the daemons in a predictable order such as: mon, mgr, osd, rgw, mds. If the Rook upgrade requires any special steps, they will be handled as each version upgrade requires.
+
+When the cluster CRD is updated with a new Ceph version, the same idempotent orchestration is executed to evaluated desired state that needs to be applied to the cluster.
+Over time as the operator becomes smarter and more versions are supported, the custom upgrade steps will be implemented as needed.
```

Comment:
Maybe adjusted this to indicate that usually a rook operator upgrade doesn't require any daemon restarts, but the ceph upgrade always does require an orderly restart of each of the pods with a new image, plus some other extra steps.

Reply:
Yes, will clarify. With the Rook upgrade, it's possible a daemon will need to be restarted if the pod spec changes, but it won't always be necessary.

Reply:
As a second option to @galexrt's above, I don't really see these design docs as snapshots of the current design state of rook but a record of the design decisions being made. Perhaps it makes sense to either date the design docs (e.g., `2018-08-30`) or mark them with a Rook version (e.g., `0.9`) being planned for. I think Rook version is easier. Then the design docs directory will show a general timeline of design decisions.